### PR TITLE
throw for now iss in uath result

### PR DESCRIPTION
**Pull request by SimonCropp**
since if we dont it is a null ref exception

### CODE REVIEW

File: src/client/Microsoft.Identity.Client/Platforms/Features/RuntimeBroker/WamAdapters.cs [hunk 1]
```diff
@@ -339,6 +339,11 @@ private static void AddPopParams(AuthenticationRequestParameters authenticationR
                 {
                     IdToken idToken = IdToken.Parse(authResult.RawIdToken);
                     authorityUrl = idToken.ClaimsPrincipal.FindFirst("iss")?.Value;
+                    if (authorityUrl == null)
+                    {
+                        throw new Exception("Expected the RawIdToken from the auth result to contain a ClaimsPrincipal named 'iss'.");
```
Comment (bgavrilMS) [inline src/client/Microsoft.Identity.Client/Platforms/Features/RuntimeBroker/WamAdapters.cs:344]: Well the `iss` claim is mandatory and this code flow is only executed for AAD which respects this.

MSAL doesn't throw generic "Exception", it throws MsalClientException or MsalServiceException. In this case, it's a client exception, but I am not sure how useful it is to an app developer to see it.

Ah well, maybe throw MsalClientException("no_iss_claim", "Id token doesn't have an iss claim in broker flow");

**Review by bgavrilMS — CHANGES_REQUESTED** [review]
Don't throw generic Exception

File: src/client/Microsoft.Identity.Client/Platforms/Features/RuntimeBroker/WamAdapters.cs (hunk 1 above)
Comment (SimonCropp) [reply src/client/Microsoft.Identity.Client/Platforms/Features/RuntimeBroker/WamAdapters.cs:344]: is iss populated by this lib? if so i think we should assume it exist. 

so

```
authorityUrl = idToken.ClaimsPrincipal.FindFirst("iss")?.Value;
```

should be 

```
authorityUrl = idToken.ClaimsPrincipal.FindFirst("iss")!.Value;
```

thoughts?

**Comment by SimonCropp** [comment]
@bgavrilMS what exception should be thrown?

**Comment by bgavrilMS** [comment]
> @bgavrilMS what exception should be thrown?

`MsalClientException`

**Comment by bgavrilMS** [comment]
It's acceptable to not throw an pretty exception in this case. closing this PR.

**Pull request closed without merging**